core/lib/libmbedtls/src/tee_lmd_provider.c

```diff
@@ -303,18 +361,95 @@ static bool bn_alloc_max(struct bignum **s)
 	return !!(*s);
 }
 
+static unsigned long int next = 1;
+
+/* Return next random integer */
+
+static int _rand(void)
+{
+	next = next * 1103515245L + 12345;
+	return (unsigned int) (next / 65536L) % 32768L;
+}
+
+static int mbd_rand(void *rng_state, unsigned char *output, size_t len)
```

Comment:
What kind of random number generator is this? How does it compare with other software based rngs? Fortuna for instance.

Reply:
@jenswi-linaro guaranteed to be very bad ;-) We don't want an LCG in OP-TEE, especially when there are better software alternatives (Fortuna as you said).

Reply:
As you know, all software rngs are from tomcrypt(may be not right), and almost all rngs from mbedtls are based on hardware. So I put this software rngs in here. If you have others function better than this, it will be better.